### PR TITLE
unskip windows build

### PR DESCRIPTION
**Pull request by mattip**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

PR #135 added support for ray-2.9.0 for linux and macos (x86_64) and disabled windows. This re-enables windows.

### CODE REVIEW

**Comment by mattip** [comment]
@conda-forge-admin, please rerender

**Comment by mattip** [comment]
I can reproduce this failure. I needed to revert much of the attempts I made in #135 to get it to build locally. Then I had to short-circuit the build cleanup to keep the build artifacts around
```diff
diff --git a/recipe/build-core.bat b/recipe/build-core.bat
index 54d0588..62a4a2d 100644
--- a/recipe/build-core.bat
+++ b/recipe/build-core.bat
@@ -22,6 +22,7 @@ echo ==========================================================

 rem remember the return code
 set RETCODE=%ERRORLEVEL%
+exit /b %RETCODE%

 rem Now clean everything up so subsequent builds (for potentially
 rem different Python version) do not stumble on some after-effects.

```

It seems the first problem is with the include paths, for some reason the bazel build of protobuf is using
```
/Iexternal/com_google_protobuf 

```

when it should be 
```
/Iexternal/com_google_protobuf/src 
```

But after that, there is this failure, and many more like it:
```
Note: including file: external/com_google_protobuf/src\google/protobuf/compiler/cpp/service.h
external/com_google_protobuf/src\google/protobuf/compiler/cpp/service.h(38): \
    fatal error C1083: Cannot open include file: 'string': No such file or directory
```

**Comment by mattip** [comment]
Properly setting `startup --output_user_root=D:/tmp` for bazel seems to have done the trick (and removing places that override that) to build ray-core. Now the error is in importing `ray.rllib`, even though the conda package is installed
```
import: 'ray.rllib'
Traceback (most recent call last):
  File "D:\bld\ray-packages_1705669809821\test_tmp\run_test.py", line 2, in <module>
    import ray.rllib
ModuleNotFoundError: No module named 'ray.rllib'
```

**Comment by mattip** [comment]
Now we are getting a failure to resolve all packages, something about pyarrow-9 and cuda. The same build on linux resolves to pyarrow-14:
```
TEST START: D:\bld\win-64\ray-data-2.9.0-py310h5588dad_3.conda
WARNING: Multiple meta files found. The meta.yaml file in the base directory (C:\Users\VSSADM~1\AppData\Local\Temp\tmprs1arz6v\info\recipe) will be used.
Reloading output folder: D:\bld
Mamba failed to solve:
 - ray-data 2.9.0.* py310h5588dad_3
 - fsspec
 - pandas >=1.3
 - pyarrow >=6.0.1
 - python >=3.10,<3.11.0a0
 - python_abi 3.10.* *_cp310
 - ray-core 2.9.0.* py310h139b6d1_3
 - python

with channels:

The reported errors are:
- Encountered problems while solving:
-   - nothing provides __cuda >=11.2 needed by pyarrow-9.0.0-py310h0d05bb6_46_cuda
- 

```

**Comment by mattip** [comment]
CI is green. Does anyone want to take a look? This re-enables windows builds by shortening the bazel path. Upstream PR https://github.com/ray-project/ray/pull/42566 is a little different but has the same effect.

File: recipe/build-core.bat [hunk 1]
```diff
@@ -14,16 +14,12 @@ echo dir %BAZEL_VC%
 dir "%BAZEL_VC%"
 
 echo ==========================================================
-echo calling bash to build
+echo calling pip to install
```
Comment (apmorton) [inline recipe/build-core.bat:17]: is this change to pip build still required? would this all work now with the original build script since patch 0006?

Comment (mattip) [reply recipe/build-core.bat:17]: The original build script used pip. I [changed it to bash](https://github.com/conda-forge/ray-packages-feedstock/pull/135/files#diff-0f49d6f77e617c54088340290b14937fdfae45eaed495ef2a679995df621b13fR16) in #135 when trying to figure out what was going on with windows.

**Review by apmorton — APPROVED** [review]

**Pull request merged**